Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -944,6 +944,25 @@ public void RaisePointerUp(IMixedRealityPointer pointer, MixedRealityInputAction
                 handler.OnInputDown(casted);
             };
 
+        private static readonly ExecuteEvents.EventFunction<IMixedRealityBaseInputHandler> OnInputDownWithActionEventHandler =
+            delegate (IMixedRealityBaseInputHandler handler, BaseEventData eventData)
+            {
+                var inputData = ExecuteEvents.ValidateEventData<InputEventData>(eventData);
+                Debug.Assert(inputData.MixedRealityInputAction != MixedRealityInputAction.None);
+
+                var inputHandler = handler as IMixedRealityInputHandler;
+                if (inputHandler != null)
+                {
+                    inputHandler.OnInputDown(inputData);
+                }
+
+                var actionHandler = handler as IMixedRealityInputActionHandler;
+                if (actionHandler != null)
```

Comment:
Do we also need to check if inputData.MixedRealityInputAction isn't **null**? I know we're checking to make sure it's not InputAction.None, but is there any point at which it could be **null**?

Reply:
`MixedRealityInputAction` is a value type so we're safe there.